homeassistant/components/tplink/sensor.py

```diff
@@ -142,54 +39,36 @@ async def async_setup_entry(
     entities: list[CoordinatedTPLinkEntity] = []
     device = parent_coordinator.device
 
-    for idx, child in enumerate(device.children):
-        # HS300 does not like too many concurrent requests and its emeter data requires
-        # a request for each socket, so we create separate coordinators.
-        if isinstance(device, SmartStrip):
-            entities.extend(
-                _async_sensors_for_device(
-                    child, children_coordinators[idx], parent=device
-                )
-            )
-        else:
-            entities.extend(
-                _entities_for_device(
-                    child,
-                    feature_type=Feature.Sensor,
-                    entity_class=Sensor,
-                    coordinator=parent_coordinator,
-                    parent=device,
-                )
-            )
-
-    entities.extend(
-        _entities_for_device(
-            device,
-            feature_type=Feature.Sensor,
-            entity_class=Sensor,
-            coordinator=parent_coordinator,
-        )
+    entities = _entities_for_device_and_its_children(
+        device=device,
+        coordinator=parent_coordinator,
+        feature_type=Feature.Type.Sensor,
+        entity_class=TPLinkSensor,
+        child_coordinators=children_coordinators,
     )
-
     async_add_entities(entities)
 
 
-class Sensor(CoordinatedTPLinkEntity, SensorEntity):
+class TPLinkSensor(CoordinatedTPLinkEntity, SensorEntity):
```

Comment:
```suggestion
class Sensor(CoordinatedTPLinkEntity, SensorEntity):
```
Let's keep the sensor name as it was. That's how `zha` is also doing, so it should be fine.

Reply:
Done in [`b53d643` (#117764)](https://github.com/home-assistant/core/pull/117764/commits/b53d6433ba36ecefd2487acdaf75c58a1e5173a0)

Reply:
You've fixed